File: Introduction to Robotics Course/Chapter12/Lesson2/Chapter12_Lesson2_Ex1.java
import example_interfaces.srv.AddTwoInts;
import org.ros2.rcljava.client.Client;

Client<AddTwoInts> cli =
  node.createClient(AddTwoInts.class, "/add");

// build request
AddTwoInts.Request req = new AddTwoInts.Request();
req.setA(3);
req.setB(7);

// async call
cli.asyncSendRequest(req).thenAccept(res -> {
  System.out.println("sum=" + res.getSum());
});
      

File: Introduction to Robotics Course/Chapter12/Lesson2/Chapter12_Lesson2_Ex1.m
% Service client for /add
cli = ros2svcclient(node, "/add", "example_interfaces/AddTwoInts");
req = ros2message(cli);
req.a = int64(3);
req.b = int64(7);

res = call(cli, req);
disp(["sum = ", num2str(res.sum)]);
      
